stix2/base.py

```diff
@@ -99,3 +101,33 @@ def __repr__(self):
         props = [(k, self[k]) for k in sorted(self._properties) if self.get(k)]
         return "{0}({1})".format(self.__class__.__name__,
                                  ", ".join(["{0!s}={1!r}".format(k, v) for k, v in props]))
+
+#  Versioning API
+
+    def new_version(self, **kwargs):
```

Comment:
How do we "unset" an attribute? Is passing `None` sufficient? Can we add a test for that?